File: prop1java/xmlprop/src/xmlsax.java
import org.xml.sax.*;
import org.xml.sax.helpers.*;
import javax.xml.parsers.*;
import java.io.File;

public class xmlsax {
    public static void main(String[] args) {
        try {
            // Crear una instancia de SAXParserFactory
            SAXParserFactory factory = SAXParserFactory.newInstance();
            // Habilitar la validación DTD
            factory.setValidating(true);

            // Crear un SAXParser
            SAXParser saxParser = factory.newSAXParser();

            // Definir el manejador de eventos
            DefaultHandler handler = new DefaultHandler() {
                boolean titulo = false;

                @Override
                public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
                    if (qName.equalsIgnoreCase("TITULO")) {
                        titulo = true;
                    }
                }

                @Override
                public void characters(char ch[], int start, int length) throws SAXException {
                    if (titulo) {
                        System.out.println("Título: " + new String(ch, start, length));
                        titulo = false;
                    }
                }

                @Override
                public void endElement(String uri, String localName, String qName) throws SAXException {
                    // Manejo del evento de fin de elemento
                }
            };

            // Parsear el archivo XML
            saxParser.parse(new File("ej01_2.xml"), handler);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: ej1Pyhton/xmlSax.py
import xml.sax

# Definir un controlador de eventos SAX
class MiControladorSAX(xml.sax.ContentHandler):
    def startElement(self, name, attrs):
        print("Elemento:", name)
        
    def characters(self, content):
        content = content.strip()
        if content:
            print("Valor:", content)

# Crear un parser SAX y asignarle el controlador de eventos
parser = xml.sax.make_parser()
handler = MiControladorSAX()
parser.setContentHandler(handler)

# Parsear un archivo XML utilizando SAX
xml_file = "boe.xml"
with open(xml_file, "r") as file:
    parser.parse(file)

